pkg/upgraders/healthcheckstep.go

```diff
@@ -40,8 +40,12 @@ func (c *clusterUpgrader) PreUpgradeHealthCheck(ctx context.Context, logger logr
 	if c.upgradeConfig.Spec.CapacityReservation {
 		ok, err := c.scaler.CanScale(c.client, logger)
 		if !ok || err != nil {
+			// HealthCheck should not fail in this senario. Have to inform the customer about the upgrade delay.
 			c.metrics.UpdateMetricHealthcheckFailed(c.upgradeConfig.Name, metrics.DefaultWorkerMachinepoolNotFound)
-			healthCheckFailed = append(healthCheckFailed, "CapacityReservationHealthcheckFailed")
+			err = c.notifier.Notify(notifier.MuoStateDelayed)
```

Comment:
This will create a SL then set the metric. If other PHC failed later, SL will not be sent out because SL sent metric has been set.

Reply:
I think SL sent metrics is based on MUO state which will be different in PHC . 
`s.metrics.UpdateMetricNotificationEventSent(uc.Name, string(state), uc.Spec.Desired.Version)`

But good to verify this once.

Reply:
I guess this might effect the UpgradeDelayCheck notification here https://github.com/openshift/managed-upgrade-operator/blob/master/pkg/upgraders/delaystep.go#L38 
Any suggestions on how to notify the user without other impact ?

Reply:
I would recommend to add an extra flag which indicate if the default worker pool check failed then adding the information accordingly.